### PR TITLE
clean up details on binary encoding syntax

### PR DESCRIPTION
**Pull request by fluffy**
This PR cleans up some nits to make all the places we use the RFC9000 encoding syntax valid, updates our extensions to it, and add a non normative summary of parts of the syntax we use. 

I don't think this is making any changes to anything on the wire or normative changes to spec other than the statement SHOULD use smallest number of bytes for variable length integers which RFC9000 only requires for some fields.

### CODE REVIEW

File: draft-ietf-moq-transport.md [hunk 1]
```diff
@@ -206,14 +206,51 @@ Track:
...
+  described in ({{?RFC9000, Section 16}}), followed by that many bytes
+  of binary data
+
+x (Location):
+
+: Indicates x is a Location structure defined in {{subscribe-locations}}.
+
+To reduce unnecessary use of bandwidth, variable length integers SHOULD
+be encoded using the least number of bytes possible to represent the
+required value.
```
Comment (fluffy) [inline draft-ietf-moq-transport.md:253]: The above sentence is the only part of this PR that seems non editorial to me. If I should put it in a separate PR, glad to do that ...

**Review by martinduke — APPROVED** [review]
I am not in love with the ABNF idea, but this is non-controversial.

**Review by suhasHere — APPROVED** [review]

File: draft-ietf-moq-transport.md (hunk 1 above)
Comment (kixelated) [reply draft-ietf-moq-transport.md:253]: `RECOMMENDED` maybe, but I would just remove this sentence IMO.

I actually used a 2-byte varint for packet sizes in QUIC so I could avoid a copy once the final size is known. I imagine you could do something similar in MoQ, or use larger varints to make rewriting OBJECT headers easier.

Either way we can't say MUST and I'm not sure what we gain with a SHOULD. A decoder has to be prepared for anything.

Comment (afrind) [reply draft-ietf-moq-transport.md:253]: Consider an RFC 6919 "OUGHT TO" https://datatracker.ietf.org/doc/html/rfc6919#section-4

Comment (fluffy) [reply draft-ietf-moq-transport.md:253]: Agree that is a real change and happy to refactor it into a separate PR.

Comment (kixelated) [reply draft-ietf-moq-transport.md:253]: > OK - updated this. I will note that from https://datatracker.ietf.org/doc/html/rfc2119#section-3 SHOULD just means "RECOMMENDED" not "REQUIRED"

Ah okay I missed that memo.

**Review by suhasHere — APPROVED** [review]

File: draft-ietf-moq-transport.md [hunk 2]
```diff
@@ -209,11 +209,47 @@ Track:
...
-: Indicates that x consists of a variable length integer (i), followed by
-  that many bytes of binary data (..).
+
+: Indicates that x consists of a variable length integer encoding as
+  described in ({{?RFC9000, Section 16}}), followed by that many bytes
+  of binary data
+
+x (Location):
+
+: Indicates x is a Location structure defined in {{subscribe-locations}}.
```
Comment (ianswett) [inline draft-ietf-moq-transport.md:248]: I find this forward reference a bit odd, and it's only used for one frame, which is likely to change soon, so I'm going to remove it to avoid it becoming out of sync.

**Review by ianswett — APPROVED** [review]
